File: src/main/java/com/flexpoker/table/command/aggregate/CommonCards.java
package com.flexpoker.table.command.aggregate;

import java.util.Arrays;
import java.util.List;

import com.flexpoker.table.command.Card;
import com.flexpoker.table.command.FlopCards;
import com.flexpoker.table.command.RiverCard;
import com.flexpoker.table.command.TurnCard;

public class CommonCards {

    private final List<Card> cards;

    public CommonCards(FlopCards flopCards, TurnCard turnCard,
            RiverCard riverCard) {
        cards = Arrays.asList(new Card[] { flopCards.getCard1(),
                flopCards.getCard2(), flopCards.getCard3(), turnCard.getCard(),
                riverCard.getCard() });
    }

    public List<Card> getCards() {
        return cards;
    }

    public FlopCards getFlopCards() {
        return new FlopCards(cards.get(0), cards.get(1), cards.get(2));
    }

    public TurnCard getTurnCard() {
        return new TurnCard(cards.get(3));
    }

    public RiverCard getRiverCard() {
        return new RiverCard(cards.get(4));
    }

}

File: src/main/kotlin/com/flexpoker/table/command/aggregate/CommonCards.kt
package com.flexpoker.table.command.aggregate

import com.flexpoker.table.command.Card
import com.flexpoker.table.command.FlopCards
import com.flexpoker.table.command.RiverCard
import com.flexpoker.table.command.TurnCard

class CommonCards(flopCards: FlopCards, turnCard: TurnCard, riverCard: RiverCard) {
    val cards: List<Card> = listOf(flopCards.card1, flopCards.card2, flopCards.card3, turnCard.card, riverCard.card)
    val flopCards: FlopCards
        get() = FlopCards(cards[0], cards[1], cards[2])
    val turnCard: TurnCard
        get() = TurnCard(cards[3])
    val riverCard: RiverCard
        get() = RiverCard(cards[4])
}